Documentation/upgrading.md

```diff
@@ -1,173 +1,149 @@
+kube-proxy                       4         4         4         4            4           <none>                            21d
+pod-checkpointer                 1         1         1         1            1           node-role.kubernetes.io/master=   21d
 
-### kube-controller-manager
+$ kubectl get deployments -n=kube-system
+kube-controller-manager           2         2         2            2           21d
+kube-dns                          1         1         1            1           21d
+kube-scheduler                    2         2         2            2           21d
+```
 
-Edit the controller-manager deployment to rolling update the controller manager. Change the container image name for the hyperkube.
+### kube-apiserver
 
-    kubectl edit deployments kube-controller-manager -n=kube-system
+Edit the `kube-apiserver` daemonset to rolling update the apiserver.
 
-Wait for the controller manager to be deployed.
+```sh
+$ kubectl edit daemonset kube-apiserver -n=kube-system
+```
 
-    $ kubectl get pods -n=kube-system
-    NAME                                       READY     STATUS    RESTARTS   AGE
-    kube-apiserver-vyg3t                       2/2       Running   0          18m
-    kube-controller-manager-1709527928-zj8c4   1/1       Running   0          4m
-    kube-dns-v20-3531996453-0tlv9              3/3       Running   0          28m
-    kube-proxy-8jthl                           1/1       Running   0          28m
-    kube-proxy-bnvgy                           1/1       Running   0          28m
-    kube-proxy-gkyx8                           1/1       Running   0          28m
-    kube-scheduler-2255275287-hti6w            1/1       Running   0          6m
-    kubelet-exe5k                              1/1       Running   0          28m
-    kubelet-p3g98                              1/1       Running   0          28m
-    kubelet-quhhg                              1/1       Running   0          28m
-    pod-checkpointer-node1.example.com         1/1       Running   0          28m
+If only the container image in a manifest has changed, update it.
 
-### Verify
+```
+kubectl set image daemonset kube-apiserver kube-apiserver=quay.io/coreos/hyperkube:v1.6.7_coreos.0
+```
 
-At this point, the control plane components have been upgraded to v1.4.3.
+If you only have one apiserver, the cluster may be momentarily unavailable.
 
-    $ kubectl version
-    Client Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.0", GitCommit:"a16c0a7f71a6f93c7e0f222d961f4675cd97a46b", GitTreeState:"clean", BuildDate:"2016-09-26T18:16:57Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
-    Server Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.3+coreos.0", GitCommit:"7819c84f25e8c661321ee80d6b9fa5f4ff06676f", GitTreeState:"clean", BuildDate:"2016-10-17T21:19:17Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
+### kube-scheduler
```

Comment:
Lets add the same image-only option for kube-scheduler and kube-controller-manager here too (but we can have an abridged note: "Again, if we are only updating the image, we can use the set image command:"